patchwork-registries/src/main/java/net/minecraftforge/registries/GameData.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Minecraft Forge, Patchwork Project
+ * Copyright (c) 2016-2020, 2019-2020
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.registries;
+
+import java.util.Locale;
+
+import org.apache.logging.log4j.LogManager;
+import net.minecraftforge.fml.ModLoadingContext;
+
+import net.minecraft.util.Identifier;
+
+public class GameData {
+	/**
+	 * Check a name for a domain prefix, and if not present infer it from the
+	 * current active mod container.
+	 *
+	 * @param name          The name or resource location
+	 * @param warnOverrides If true, logs a warning if domain differs from that of
+	 *                      the currently currently active mod container
+	 *
+	 * @return The {@link net.minecraft.util.Identifier} with given or inferred domain
+	 */
+	public static Identifier checkPrefix(String name, boolean warnOverrides) {
+		int index = name.lastIndexOf(':');
+		String oldPrefix = index == -1 ? "" : name.substring(0, index).toLowerCase(Locale.ROOT);
+		name = index == -1 ? name : name.substring(index + 1);
+		String prefix = ModLoadingContext.get().getActiveNamespace();
+
+		if (warnOverrides && !oldPrefix.equals(prefix) && oldPrefix.length() > 0) {
+			LogManager.getLogger().info("Potentially Dangerous alternative prefix `{}` for name `{}`, expected `{}`. This could be a intended override, but in most cases indicates a broken mod.", oldPrefix, name, prefix);
+			prefix = oldPrefix;
+		}
+
+		return new Identifier(prefix, name);
```

Comment:
There's a version of this code in `ExtendedForgeRegistryEntry` already, it would be appreciated if you could deduplicate it.

Reply:
You mean make ForgeRegistryEntry extend ExtendedForgeRegistryEntry?

Reply:
That would be good too, but in this case it would be nice to move the logic to a helper class outside of GameData and ExtendedForgeRegistryEntry.

Reply:
I found that making ForgeRegistryEntry extend ExtendedForgeRegistryEntry is wrong because it turns class method into interface method. And GameData is already a helper class. ExtendedForgeRegistryEntry.setRegistryName unlike this one does not replace the namespace so these two cannot be generalized.

Reply:
oh it can be generalized but it will make the code weird